models/evc.py

```diff
@@ -1154,35 +1154,36 @@ def check_trace(
-        trace_a = circuit_by_traces[circuit_id]['trace_a']
-        trace_z = circuit_by_traces[circuit_id]['trace_z']
-        if len(trace_a) != len(circuit_current_path) + 1:
-            log.warning(f"Invalid trace from uni_a: {trace_a}")
-            circuits_checked[circuit_id] = False
-        if len(trace_z) != len(circuit_current_path) + 1:
-            log.warning(f"Invalid trace from uni_z: {trace_z}")
-            circuits_checked[circuit_id] = False
-        for link, trace1, trace2 in zip(circuit_current_path,
-                                        trace_a[1:],
-                                        trace_z[:0:-1]):
-            metadata_vlan = None
-            if link.metadata:
-                metadata_vlan = glom(link.metadata, 's_vlan.value')
-            if compare_endpoint_trace(
-                                        link.endpoint_a,
-                                        metadata_vlan,
-                                        trace2
-                                    ) is False:
+        if circuit_current_path:
+            circuits_checked[circuit_id] = True
+            trace_a = circuit_by_traces[circuit_id]['trace_a']
+            trace_z = circuit_by_traces[circuit_id]['trace_z']
+            if len(trace_a) != len(circuit_current_path) + 1:
                 log.warning(f"Invalid trace from uni_a: {trace_a}")
                 circuits_checked[circuit_id] = False
-            if compare_endpoint_trace(
-                                        link.endpoint_b,
-                                        metadata_vlan,
-                                        trace1
-                                    ) is False:
+            if len(trace_z) != len(circuit_current_path) + 1:
                 log.warning(f"Invalid trace from uni_z: {trace_z}")
                 circuits_checked[circuit_id] = False
+            for link, trace1, trace2 in zip(circuit_current_path,
+                                            trace_a[1:],
+                                            trace_z[:0:-1]):
+                metadata_vlan = None
+                if link.metadata:
+                    metadata_vlan = glom(link.metadata, 's_vlan.value')
+                if compare_endpoint_trace(
+                                            link.endpoint_a,
+                                            metadata_vlan,
+                                            trace2
+                                        ) is False:
+                    log.warning(f"Invalid trace from uni_a: {trace_a}")
+                    circuits_checked[circuit_id] = False
+                if compare_endpoint_trace(
+                                            link.endpoint_b,
+                                            metadata_vlan,
+                                            trace1
+                                        ) is False:
+                    log.warning(f"Invalid trace from uni_z: {trace_z}")
+                    circuits_checked[circuit_id] = False
```

Comment:
@gretelliz, in the changes above, could you please check if it is possible just add an early return instead of adding all the functions to the IF clause? Something like:

```
if not circuit_current_path:
     return
```

Reply:
@gretelliz I suggested a minor adjustment proposed above in commit 7dbf9de, simplifying the changes on this PR. Please let me know if you are okay with that.